### DISCUSSION TITLE
Is it possible to use the iCub arm without the hand?

### DISCUSSION THREAD

**Discussion by S-Dafarra** [robotology/community#213] [Q&A] [answered]
I am interested in using the robot arm without the hand. In practice I may need to use the shoulder and the elbow even if the hand have some problems at startup. Is it possible? What is the cleanest way to do so?

@DanielePucci  @pi-q @gabrielenava  @traversaro @randaz81

**Comment by S-Dafarra** [comment, marked as answer]
Actually I have found a method, but I 'm not sure it is the cleanest one. 
First of all I started by modifying the ``yarpwholebodyinterface.ini`` file located in the ```.local``` folder in the iCub head, like in https://github.com/S-Dafarra/robots-configuration/commit/87f5495bbba3c62fe24a70f44edf80ad5f11625e .
Then I modified the [icub_all.xml](https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/icub_all.xml) file, by copying its content into a file called ```icub_no_right_hand.xml```, with the following modifications:
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/icub_no_right_hand.xml#L29-L33
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/icub_no_right_hand.xml#L53
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/icub_no_right_hand.xml#L59-L60
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/icub_no_right_hand.xml#L92 .

Similarly I copied the content of the file [wrappers/motorControl/right_arm-mc_wrapper.xml](https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/motorControl/right_arm-mc_wrapper.xml) into ```wrappers/motorControl/right_arm_no_hand-mc_wrapper.xml``` by performing the following modifications:
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/motorControl/right_arm_no_hand-mc_wrapper.xml#L10-L12
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/motorControl/right_arm_no_hand-mc_wrapper.xml#L23
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/motorControl/right_arm_no_hand-mc_wrapper.xml#L23

Then I modified the [/wrappers/VFT/right_arm-VFT_wrapper.xml](https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/VFT/right_arm-VFT_wrapper.xml) file with the new name ```/wrappers/VFT/right_arm_no_hand-VFT_wrapper.xml``` with the following modifications:
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/VFT/right_arm_no_hand-VFT_wrapper.xml#L12
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/wrappers/VFT/right_arm_no_hand-VFT_wrapper.xml#L22

Finally I edited the [/calibrators/right_arm-calib.xml](https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/calibrators/right_arm-calib.xml) file into ```/calibrators/right_arm_no_hand-calib.xml``` by doing:
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/calibrators/right_arm_no_hand-calib.xml#L9
- https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/calibrators/right_arm_no_hand-calib.xml#L35
- Removing all the columns from 4 to the end like in lines https://github.com/S-Dafarra/robots-configuration/blob/0c17f1fe232faad9eb1b239996696c0acc6526c7/iCubGenova04/calibrators/right_arm_no_hand-calib.xml#L12-L33 

As I said I don't think this is the cleanest way to do so, neither the easiest, but maybe it is a solution. Hope this to be useful.

**Comment by DanielePucci** [comment]
👍
